tests/tests.rs

```diff
@@ -269,6 +269,99 @@ fn reject_table_get_set() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn reject_table_get_set_with_reference_types_enabled() -> Result<()> {
+    let wat = r#"
+      (module
+        (table 3 funcref)
+
+        (func $f (result i32) (i32.const 0))
+        (func $g (result i32) (i32.const 0))
+        (func $h (result i32) (i32.const 0))
+
+        (func (export "main")
+          i32.const 0
+          i32.const 1
+          table.get
+          table.set)
+
+        (elem (i32.const 0) $f $g $h)
+      )"#;
+
+    let _ = env_logger::try_init();
+    let mut wizer = Wizer::new();
+    wizer.wasm_reference_types(true);
+
+    let wasm = wat_to_wasm(wat)?;
+    let result = wizen_and_run_wasm(&[], 42, &wasm, wizer);
+
+    assert!(result.is_err());
+
+    let err = result.unwrap_err();
+    assert!(err
+        .to_string()
+        .contains("unsupported `table.get` instruction"),);
+
+    Ok(())
+}
+
+#[test]
+fn reject_table_grow_with_reference_types_enabled() -> anyhow::Result<()> {
+    let wat = r#"
+      (module
+        (elem declare func $f)
+        (func $f)
+        (table 0 funcref)
+        (func (export "_initialize") (result i32)
+            ref.func $f
+            i32.const 1
+            table.grow
+        )
+      )"#;
+
+    let _ = env_logger::try_init();
+    let mut wizer = Wizer::new();
+    wizer.wasm_reference_types(true);
+
+    let wasm = wat_to_wasm(wat)?;
+    let result = wizen_and_run_wasm(&[], 42, &wasm, wizer);
+
+    assert!(result.is_err());
+
+    let err = result.unwrap_err();
+    assert!(err
+        .to_string()
+        .contains("unsupported `ref.func` instruction"));
+
+    Ok(())
+}
+
+#[test]
+fn indirect_call_with_reference_types() -> anyhow::Result<()> {
+    let wat = r#"
+      (module
+        (type $sig (func (result i32)))
+        (table 1 funcref)
+        (elem (i32.const 0) $f)
+        (func $f (type $sig)
+          i32.const 42
+        )
+        (func (export "wizer.initialize"))
+        (func (export "run") (result i32)
+          i32.const 0
+          call_indirect (type $sig)
+        )
+      )"#;
```

Comment:
I don't think that this WAT is actually exercising the `call_indirect` instruction encoding that was added in reference-types. Can you make two tables, a dummy table that is unused first, and then this current table second, and then add `(table 1)` to the `call_indirect` so that it is not using the default `(table 0)`, which requires newer encodings? Thanks!